spec/maths/statistics/vector_spec.rb

```diff
@@ -693,24 +693,43 @@
     end
   end
 
-  context "#macd" do
-    it "calculates moving average convergence divergence" do
-      # MACD uses a lot more data than the other ones, so we need a bigger vector
-      data = Daru::Vector.new(
-        File.readlines("spec/fixtures/stock_data.csv").map(&:to_f))
+  RSpec.shared_examples 'correct macd' do |*settings|
+    let(:source) { Daru::DataFrame.from_csv('spec/fixtures/macd_data.csv') }
 
-      macd, signal = data.macd
+    # skip initial records during compare as ema is sensitive to
+    # period used.
+    # http://ta-lib.org/d_api/ta_setunstableperiod.html
+    let(:stability_offset) { 90 }
+    let(:delta) { 0.001 }
 
-      # check the MACD
-      expect(macd[-1]).to be_within(1e-6).of(3.12e-4)
-      expect(macd[-10]).to be_within(1e-4).of(-1.07e-2)
-      expect(macd[-20]).to be_within(1e-5).of(-5.65e-3)
+    if settings.empty?
+      subject { source['price'].macd }
+      settings_str = '12_26_9'
+    else
+      fast, slow, signal = settings
+      subject { source['price'].macd fast, slow, signal }
+      settings_str = settings.join('_')
+    end
+
+    %w[ macd macdsig macdhist ].each_with_index do |field, i|
+      macd_type = "#{field}_#{settings_str}"
```

Comment:
No need in this local variable, it is ugly and used only once in code below, so could easily be inlined.